### PR TITLE
Abstraction of "reading in file" to the top level class

### PR DESCRIPTION
**Pull request by thomasyu888**
No need to read in data frame for every single class. This should be the default.

### CODE REVIEW

File: processing/example_filetype_format.py [hunk 1]
```diff
@@ -51,16 +55,30 @@ def process(self, filePath, *args, **kwargs):
...
-	def validate_steps(self, filePathList, **kwargs):
-		total_error = ""
+	def _validate(self, df):
+		total_error =""
 		warning = ""
+		logger.info("NO VALIDATION for %s files" % self._fileType)
 		return(total_error, warning)
 
+	def _call_validate(self, df, **kwargs):
+		return(self._validate(df))
```
Comment (kkdang) [inline processing/example_filetype_format.py]: Comment about getting kwargs here to pass into _validate

File: genie/vcf.py [hunk 2]
```diff
@@ -29,7 +29,19 @@ class vcf(maf.maf):
 
 	def _validateFilename(self, filePath):
 		assert os.path.basename(filePath[0]).startswith("GENIE-%s-" % self.center) and os.path.basename(filePath[0]).endswith(".vcf")
-	
+
+	def readFile(self, filePathList):
+		headers = None
+		with open(filePath,"r") as foo:
```
Comment (thomasyu888) [inline genie/vcf.py]: filePathList

File: genie/example_filetype_format.py [hunk 3]
```diff
@@ -17,7 +18,11 @@ def __init__(self, syn, center, poolSize=1):
 		self.syn = syn
 		self.center = center
 		#self.pool = multiprocessing.Pool(poolSize)
-
+	
+	def readFile(self, filePathList):
+		filePath = filePathList[0]
+		df = pd.read_csv(filePath,sep="\t",comment="#")
```
Comment (thomasyu888) [inline genie/example_filetype_format.py:22]: get_dataframe function

File: genie/example_filetype_format.py [hunk 4]
```diff
@@ -51,16 +56,30 @@ def process(self, filePath, *args, **kwargs):
...
 
-	def validate_steps(self, filePathList, **kwargs):
-		total_error = ""
+	def _validate(self, df):
+		total_error =""
 		warning = ""
+		logger.info("NO VALIDATION for %s files" % self._fileType)
 		return(total_error, warning)
 
+	def _call_validate(self, df, **kwargs):
```
Comment (thomasyu888) [inline genie/example_filetype_format.py]: Remove _call_validate...

**Review by zimingd — APPROVED** [review]

**Pull request merged**